Program.cs

```diff
@@ -0,0 +1,238 @@
+﻿using RestSharp;
+using Newtonsoft.Json;
+using System.Data.SQLite;
+namespace NewDev;
+
+public class AuthResponse 
+{
+    public string Id { get; set; }
+    public int Code { get; set; }
+    public string Message { get; set; }
+    public AuthResponseData? Data { get; set; }
+}
+
+public class AuthResponseData
+{
+    public int Id { get; set; }
+    public string Name { get; set; }
+    public string Email { get; set; }
+    public string Token { get; set; }
+}
+
+public class UsersResponse
+{
+    public string Id { get; set; }
+    public int Page { get; set; }
+    public int Per_Page { get; set; }
+    public int Totalrecord { get; set; }
+    public int Total_Pages { get; set; }
+    public List<User> Data { get; set; }
+}
+
+public class User
+{
+    public int Id { get; set; }
+    public string Name { get; set; }
+    public string Email { get; set; }
+    public string ProfilePicture { get; set; }
+    public string Location { get; set; }
+    public string Createdat { get; set; }
+}
+
+public class Program 
+{
+    private static string? AuthToken { get; set; }
+
+    // dotnet run register -u <email> -p <password>
+    public static void Main(string[] args) 
+    {
+        // if user table exists, ignore it
+        CreateTable();
+        
+        // Api Url = http://restapi.adequateshop.com
+        var url = "http://restapi.adequateshop.com/api";
+        RestClient client = new(url);
+
+        // 1: Allow the user to input an email and password (command-line entry)
+        Console.WriteLine("Register or Login?: (Input R or L)");
+        var loginRegister = Console.ReadLine();
+
+        Console.WriteLine("\n\nEnter Email: ");
+        var email = Console.ReadLine();
+
+        Console.WriteLine("Enter Password: ");
+        var password = Console.ReadLine();
+        
+        RestResponse response;
+
+        // 2: Register || Login user into the api
+        if (loginRegister == "R") {
```

Comment:
The logic here was a bit disordered and could have been abstracted via private helper so that it would read better. 

What I could have done was take the inputs entered by the user as a dictionary with keys being the loginRegister, and fields dependent on that  and return it to make the branching a bit easier to understand.